Extensions/dnSpy.StringSearcher/StringReferencesService.cs

```diff
@@ -0,0 +1,228 @@
+
+		[ImportingConstructor]
+		public StringReferencesService(
+			IDecompilerService decompilerService,
+			ITextElementProvider textElementProvider,
+			IClassificationFormatMapService classificationFormatMapService,
+			IDocumentTabService documentTabService,
+			IMenuService menuService,
+			IWpfCommandService wpfCommandService,
+			IDotNetImageService dotNetImageService) {
+
+			this.decompilerService = decompilerService;
+			this.textElementProvider = textElementProvider;
+			this.classificationFormatMapService = classificationFormatMapService;
+			this.documentTabService = documentTabService;
+			this.dotNetImageService = dotNetImageService;
+			UIObject = new StringsControl {
+				DataContext = vm = new StringsControlVM(this)
+			};
+
+			dispatcher = Dispatcher.CurrentDispatcher;
+
+			addItems = items => {
+				foreach (var item in items) {
+					vm.StringLiterals.Add(item);
+				}
+			};
+
+			var listboxGuid = new Guid(StringSearcherConstants.GUID_STRINGS_LISTBOX);
+			menuService.InitializeContextMenu(
+				UIObject.ListView,
+				listboxGuid,
+				new GuidObjectsProvider()
+			);
+
+			wpfCommandService.Add(listboxGuid, UIObject.ListView);
+
+			var commands = wpfCommandService.GetCommands(listboxGuid);
+			commands.Add(new RelayCommand(_ => FollowSelectedReference(false)), ModifierKeys.None, Key.Enter);
+			commands.Add(new RelayCommand(_ => FollowSelectedReference(true)), ModifierKeys.Control, Key.Enter);
+			commands.Add(new RelayCommand(_ => FollowSelectedReference(true)), ModifierKeys.Shift, Key.Enter);
+
+			UIObject.ListView.MouseDoubleClick += (_, _) => {
+				FollowSelectedReference((Keyboard.Modifiers & ModifierKeys.Control) != 0);
+			};
+		}
+
+		private void FollowSelectedReference(bool newTab) {
+			if (vm.SelectedStringReference is { } selected) {
+				FollowReference(selected, newTab);
+			}
+		}
+
+		public void Analyze(IEnumerable<ModuleDef> modules) {
+			selectedModules = modules.ToArray();
+			AnalyzeSelectedModules();
+		}
+
+		public void Analyze(ModuleDef module) => Analyze([module]);
+
+		private void AnalyzeSelectedModules() {
+			var context = new StringReferenceContext(
+				decompilerService.Decompiler,
+				textElementProvider,
+				classificationFormatMapService.GetClassificationFormatMap("UIMisc"), // TODO: replace string with AppearanceCategoryConstants.UIMisc
+				dotNetImageService
+			);
+
+			vm.StringLiterals.Clear();
+
+			Task.Factory.StartNew(() => {
+				Parallel.ForEach(selectedModules.SelectMany(x => x.GetTypes()), type => {
+					var items = new List<StringReference>();
+
+					if (type.HasFields)
+						AnalyzeConstantProviders(context, type.Fields, items);
+
+					if (type.HasProperties)
+						AnalyzeConstantProviders(context, type.Properties, items);
+
+					foreach (var method in type.Methods) {
+						if (method.HasParamDefs)
+							AnalyzeConstantProviders(context, method.ParamDefs, items);
+						if (method.HasBody && method.Body is { HasInstructions: true })
```

Comment:
Checking the property `HasBody` is redundant here, the pattern matching right after will check if `Body` is null.